internal/client/client.go

```diff
@@ -14,12 +14,34 @@ import (
 	"golang.org/x/oauth2"
 	"golang.org/x/oauth2/clientcredentials"
 
+	"encoding/base64"
+	"encoding/json"
+
 	"github.com/auth0/go-auth0"
 )
 
 // UserAgent is the default user agent string.
 var UserAgent = fmt.Sprintf("Go-Auth0-SDK/%s", auth0.Version)
 
+// Telemetry is the structure used to send telemetry data in the "Auth0-Client" header.
+type Telemetry struct {
```

Comment:
While some of our older SDKs still call this Telemetry, more recently we've started to call it UserAgent instead, as "telemetry" can often have negative connotations and raise eyebrows.
For example: https://github.com/auth0/Auth0.Android/pull/398

Please consider renaming the feature.

Reply:
Other example:
https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/lib/auth0_flutter.dart#L27

Reply:
I think that's a fair point, unfortunately this package already exposes a [configurable user agent](https://github.com/auth0/go-auth0/blob/main/internal/client/client.go#L21) so I'm not sure I can rename it to that, in SPA-JS it's called [`Auth0Client`](https://auth0.github.io/auth0-spa-js/interfaces/Auth0ClientOptions.html#auth0Client), what do you think about that?

Reply:
Is that user agent string public? If not, you can change that as well. Otherwise, the SPA SDK solution is fin 👍🏼

Reply:
Configuring the  user agent is public, by calling `WithUserAgent` on a `management.New` call and is used by [auth0 cli](https://github.com/auth0/auth0-cli/blob/e4cab13cebf1b949b15c018d36357663db609a26/internal/cli/cli.go#L212) and [terraform](https://github.com/auth0/terraform-provider-auth0/blob/f11337f3818810e9fde04be184b46ff3f623b855/internal/provider/provider.go#L133) currently

Reply:
Pushed c336a83a92ef478bbe2e7de232ed68a0751f5431 to change this, I ending up using `Auth0ClientInfo` to try and convey that this is information about the client rather than a client that can be used to call API methods (if that makes sense)